SPECS/xapi.spec

```diff
@@ -862,6 +877,18 @@ systemctl start wsproxy.socket >/dev/null 2>&1 || :
 %posttrans -n message-switch
 /bin/systemctl daemon-reload >/dev/null 2>&1 || :
 
+%posttrans -n xcp-rrdd
+## On upgrade, the plugin protocol can change, so shut down
+## existing plugins and start new ones.
+## NOTE: This will actually restart plugins on *any* kind of transaction involving xcp-rrdd, not only
+## upgrades. This is because there is no way to distinguish upgrades from other kinds of transactions
+## in RPM < 4.12, see https://pagure.io/packaging-committee/issue/1051.
```

Comment:
Note for XS packagers: you can detect whether it's an upgrade in `%post`, put a temporary file in `/tmp` and then act in `%posttrans`  when the file is present (and of course, delete the file for next time).

See the `%regenerate_initrd_post` and `%regenerate_initrd_posttrans` macros regarding this "drop a file to tell %posttrans to act" thing.

This would be preferrable to needlessly attempting to restart the plugins in a chroot when installing the system.